src/main/java/net/juniper/netconf/NetconfSession.java

```diff
@@ -151,7 +153,20 @@ String getRpcReply(String rpc) throws IOException {
     private BufferedReader getRpcReplyRunning(String rpc) throws IOException {
         // RFC conformance for XML type, namespaces and message ids for RPCs
         messageId++;
-        rpc = rpc.replace("<rpc>", "<rpc xmlns=\"" + NetconfConstants.URN_XML_NS_NETCONF_BASE_1_0 + "\" message-id=\"" + messageId + "\">").trim();
+        if(rpcAttributes.isEmpty())
+            rpc = rpc.replace("<rpc>", "<rpc xmlns=\"" + NetconfConstants.URN_XML_NS_NETCONF_BASE_1_0 + "\" message-id=\"" + messageId + "\">").trim();
+        else {
+            StringBuilder attributes = new StringBuilder();
+            boolean defaultNamespaceSet = false;
+            for (Map.Entry<String, String> attribute : rpcAttributes.entrySet()) {
+                attributes.append(String.format(" %1s=\"%2s\"", attribute.getKey(), attribute.getValue()));
+                if (attribute.getKey().equals("xmlns"))
+                    defaultNamespaceSet = true;
+            }
+            if (!defaultNamespaceSet)
+                attributes.append(" xmlns=\""+NetconfConstants.URN_XML_NS_NETCONF_BASE_1_0+"\"");
```

Comment:
missed code formatting

Reply:
it supposed to be with more spaces inside. like:
```
attributes.append(" xmlns=\"" + NetconfConstants.URN_XML_NS_NETCONF_BASE_1_0 + "\"");
```
what IDE do you use? IDE usually can do it automatically

Reply:
I use IntelliJ. I'm not sure why it didn't properly format this the first time around, but I have made this correction now.